server/src/grpc/server.rs

```diff
@@ -112,31 +117,150 @@ impl RpcService for ServerImpl {
+        return;
+    };
+
+    match client.update_waypoints(request).await {
+        Ok(response) => {
+            grpc_info!("RESPONSE={:?}", response);
+        }
+        Err(e) => {
+            grpc_error!("ERROR={:?}", e);
+        }
+    }
+}
+
+/// Periodically pulls down waypoints from the regional interface and
+///  pushes them to the GIS microservice
+pub async fn waypoints_loop(config: Config, region: Box<dyn RegionInterface + Send + Sync>) {
+    let host = config.gis_host_grpc;
+    let port = config.gis_port_grpc;
+
+    grpc_info!(
+        "(waypoints_loop) Starting loop with interval: {} seconds.",
+        config.interval_seconds_refresh_waypoints
+    );
+
+    let mut cache: HashMap<String, Coordinates> = HashMap::new();
+
+    loop {
+        // Pull down waypoints from regional interface
+        region.acquire_waypoints(&mut cache).await;
+        update_waypoints(host.clone(), port, &cache).await;
+        std::thread::sleep(std::time::Duration::from_secs(
+            config.interval_seconds_refresh_waypoints as u64,
+        ));
+    }
+}
+
+async fn update_restrictions(
+    host: String,
+    port: u16,
+    restrictions: &HashMap<String, RestrictionDetails>,
+) {
+    let mut zones: Vec<NoFlyZone> = vec![];
+    for (label, details) in restrictions.iter() {
+        let time_start = match details.timestamp_start {
+            Some(t) => match datetime_to_timestamp(&t) {
+                Some(t) => Some(t),
+                _ => {
+                    grpc_error!("(restrictions_loop) could not convert timestamp for zone with label {label}.");
+                    continue;
+                }
+            },
+            None => None,
+        };
+
+        let time_end = match details.timestamp_end {
+            Some(t) => match datetime_to_timestamp(&t) {
+                Some(t) => Some(t),
+                _ => {
+                    grpc_error!("(restrictions_loop) could not convert timestamp for zone with label {label}.");
+                    continue;
+                }
+            },
+            None => None,
+        };
+
+        zones.push(NoFlyZone {
+            label: label.clone(),
+            vertices: details
+                .vertices
+                .iter()
+                .map(|v| Coordinates {
+                    latitude: v.latitude,
+                    longitude: v.longitude,
+                })
+                .collect(),
+            time_start,
+            time_end,
+        });
+    }
+
+    if zones.is_empty() {
+        grpc_warn!("(restrictions_loop) no restrictions to update.");
+        return;
+    }
+
+    let request = tonic::Request::new(UpdateNoFlyZonesRequest { zones });
+    let address = format!("{}:{}", host, port);
+    let result = GisClient::connect(address).await;
+    let Ok(mut client) = result else {
+        grpc_error!("Failed to connect to gRPC server: {}", result.unwrap_err());
```

Comment:
Missing function name in log message

Reply:
https://github.com/Arrow-air/svc-compliance/pull/11/commits/87b60b28c50e34eac6a06a06f1ae6476b4e291c1